components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +325,79 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  private showInputs(fieldName: string): void {
```

Comment:
```suggestion
  private showInputs(fieldName: FieldName): void {
```
then define this literal string type
```
export type FieldName = 'type' | 'identity' | 'name';
```
...just above the class definition in the file.

Types in TypeScript are... "strong-ish." This change does not actually add as much benefit as you might think. For instance, that correction I suggested to the unit test (using "team" instead of "type") would still not be noticed.
But it does provide two small but important benefits.
1. A reader of this code will know precisely what inputs are supposed to be fed to showInputs; with type "string" as in L329 one has no idea of the constraints.
2. This will generate compile-time errors if a typo is introduced on L337 or L345.

Please see _TypeScript Type Traps_ at https://docs.google.com/presentation/d/1ir7P7V0tzO4cd4vCtyUb-a5fHFnBsRYTUoCWLq6xksU/edit#slide=id.g20011f5b3f_0_0

Reply:
Thanks for this explanation!